django/core/files/storage/memory.py

```diff
@@ -0,0 +1,231 @@
+import errno
+import io
+import os
+from urllib.parse import urljoin
+
+from django.conf import settings
+from django.core.files.base import ContentFile
+from django.core.signals import setting_changed
+from django.utils.deconstruct import deconstructible
+from django.utils.encoding import filepath_to_uri
+from django.utils.functional import cached_property
+from django.utils.timezone import now
+
+from .base import Storage
+from .mixins import StorageSettingsMixin
+
+__all__ = ("InMemoryStorage",)
+
+
+class InMemoryFileNode(ContentFile):
+    """
+    Helper class representing an in-memory file node
+    """
+
+    def __init__(self, content="", name=""):
+        self.file = None
+        self._content_type = type(content)
+        self._initialize_stream()
+
+        self.created_time = now()
+        self.accessed_time = now()
+        self.modified_time = now()
```

Comment:
Should we call `now()` three times?
```suggestion
        self.created_time = now()
        self.accessed_time = self.created_time
        self.modified_time = self.created_time
```

As far as I'm aware all dates should be the same when a file is created.